### PR TITLE
Add normalized mutual information metric

### PR DESCRIPTION
**Pull request by jni**
## Description

I decided to break #3544 into several smaller PRs that are easier to review. This is the first one and adds normalized mutual information to simple metrics. Should be useful in various contexts.

## Checklist

- [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- Gallery example in `./doc/examples` (new features only)
- Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- Unit tests
- Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)

## For reviewers

- Check that the PR title is short, concise, and will make sense 1 year
  later.
- Check that new functions are imported in corresponding `__init__.py`.
- Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.

### CODE REVIEW

**Comment by emmanuelle** [comment]
I think you forgot to update or add the `__init__.py`.

File: skimage/metrics/simple_metrics.py
Comment (emmanuelle) [inline skimage/metrics/simple_metrics.py:187]: Just greater I think, not "strictly greater" ?

File: skimage/metrics/tests/test_simple_metrics.py
Comment (emmanuelle) [inline skimage/metrics/tests/test_simple_metrics.py:104]: Could you please add a test with a 3d image and different sizes to be sure that it works also in 3d? (It should from what I read).

**Comment by emmanuelle** [comment]
Thanks for the PR! I left a couple of minor comments.

**Comment by jni** [comment]
Thanks @emmanuelle, and oops! 😅 I've addressed all your comments. =)

**Review by emmanuelle — APPROVED** [review]
Thanks @jni! Now we need a second review :-).

File: skimage/metrics/simple_metrics.py
Comment (mkcor) [inline skimage/metrics/simple_metrics.py:10]:
```suggestion
           'normalized_mutual_information',
```

File: skimage/metrics/simple_metrics.py
Comment (mkcor) [inline skimage/metrics/simple_metrics.py:196]:
```suggestion
    The normalized mutual information of :math:`A` and :math:`B` is given by::
```

File: skimage/metrics/simple_metrics.py
Comment (mkcor) [inline skimage/metrics/simple_metrics.py:202]:
```suggestion
    where :math:`H(X) := - \sum_{x \in X}{x \log x}` is the entropy.
```

File: skimage/metrics/simple_metrics.py
Comment (mkcor) [inline skimage/metrics/simple_metrics.py:194]:
```suggestion
    r"""Compute the normalized mutual information (NMI).
```
(because these initials appear later on)

File: skimage/metrics/simple_metrics.py
Comment (mkcor) [inline skimage/metrics/simple_metrics.py:187]:
```suggestion
        raise ValueError(f"Target shape {shape} must be greater than or equal to input"
```
`>`: greater than
`≥`: greater than or equal to
Alternatively, "cannot be less than input shape..."

**Review by mkcor — COMMENTED** [review]
Thanks for the addition!

File: skimage/metrics/simple_metrics.py
Comment (jni) [inline skimage/metrics/simple_metrics.py:188]:
```suggestion
        raise ValueError(f"Target shape {shape} cannot be smaller than input"
                         f"shape {arr.shape} along any axis.")
```

**Comment by jni** [comment]
Thanks for the review @mkcor! I've applied your suggestions.

**Review by grlee77 — APPROVED** [review]
Thanks for breaking up the PR into more manageable pieces. The implementation looks consistent with the cited reference.

File: skimage/metrics/simple_metrics.py
Comment (rfezzani) [inline skimage/metrics/simple_metrics.py:144]: `im_true` must be `image_true` :smile:

File: skimage/metrics/simple_metrics.py
Comment (rfezzani) [inline skimage/metrics/simple_metrics.py:193]: For consistency with `normalized_root_mse` and `peak_signal_noise_ratio`, may be `im_true` and `im_test` musr be renamed to `image_true` and `image_test`.

BTW, I see that `mean_squared_error` uses `image0` and `image1` :confused:, this can be a new PR :wink:.

File: skimage/metrics/simple_metrics.py
Comment (rfezzani) [inline skimage/metrics/simple_metrics.py:252]:
```suggestion
    hist, bin_edges = np.histogramdd([padded_true.reshape(-1),
                                      padded_test.reshape(-1)],
```

File: skimage/metrics/simple_metrics.py
Comment (rfezzani) [inline skimage/metrics/simple_metrics.py:257]:
```suggestion
    H_true_test = entropy(hist.reshape(-1))
```

**Review by rfezzani — COMMENTED** [review]
Thank you @jni, I simply suggested a variable renaming and some minor changes :wink:

File: skimage/metrics/simple_metrics.py
Comment (jni) [reply skimage/metrics/simple_metrics.py:193]: I think we decided that symmetric measures would use image0 and image1, and asymmetric ones would use image_true and image_test. So I think that argues that these should be image0 and image1?

Comment (jni) [reply skimage/metrics/simple_metrics.py:193]: CC @scikit-image/core

Comment (stefanv) [reply skimage/metrics/simple_metrics.py:193]: Interesting! Perhaps we can simply adopt the true/test for both, even though it is slightly odd for the prior case. I have no strong intuition here.

Comment (jni) [reply skimage/metrics/simple_metrics.py:193]: LOL I hunted down the [original discussion](https://github.com/scikit-image/scikit-image/pull/4187#issuecomment-537842032), which happened just over a year ago, and contains this nugget:

> After discussing with Stéfan in the community meeting this week, we settled on the following:
>
> - for comparison functions, `image0` and `image1` where the metric is in a strong sense symmetric, and `image_true` and `image_test` where it is not. (though this could be `true_image` and `test_image`, come to think of it)

😂

Comment (mkcor) [reply skimage/metrics/simple_metrics.py:193]: Interesting, indeed! I'm in favour of sticking with what you folks settled a year ago, so `image0` and `image1` in this case.

**Comment by emmanuelle** [comment]
@jni let's get this one in if you have the bandwidth these days!

File: skimage/metrics/simple_metrics.py
Comment (stefanv) [reply skimage/metrics/simple_metrics.py:193]: Given that these are not named keywords, I think it's fine to use the argument name to indicate the nature of what's happening.  But, again, I don't feel strongly either way---as long as the documentation is clear, this shouldn't be too confusing.

**Comment by jni** [comment]
Ok, I think I've addressed all review comments! Thanks for all the catches, everyone! 😊

**Comment by rfezzani** [comment]
Excellent! Thank you @jni :tada:

**Pull request merged**
